### PR TITLE
Add note to docs explaining the use of `"validation"` vs `"parse"`

### LINKED ISSUE #3884: Docs should be consistent on whether validation is a primary purpose or not
An infobox in the [docs on models](https://pydantic-docs.helpmanual.io/usage/models/) points out that (emphasis not mine)
>pydantic is primarily a parsing library, **not a validation library**.

Simultaneously, the *very first* sentences (taglines) of both the README the docs say
>Data validation and settings management [...] Define how data should be in pure, canonical Python 3.7+; validate it with pydantic.

This is confusing and one is tempted to conclude that either those intros or that infobox must be outdated, feel compelled to spend time looking into which one it is, eventually coming across #578 where some comments (https://github.com/samuelcolvin/pydantic/issues/578#issuecomment-533256681, https://github.com/samuelcolvin/pydantic/issues/578#issuecomment-708359477) already pointed out this exact issue.

Something that would, in my opinion, help a bit against this particular kind of confusion would be small additions to the infobox's exact phrasing, e.g. just acknowledging the contradiction:
>Despite its tagline, pydantic is actually primarily a parsing library, not a validation library.

### PR DESCRIPTION
**Pull request by sydney-runkle**
## Change Summary

Adding a note explaining why we've decided to switch to the use of "validation" over "parse", except in a json parsing context. Removing any confusion about "validation" not being a primary purpose of the library.

## Related Issues

Fix #3884

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Please review

**Review by hramezani — APPROVED** [review]

File: docs/concepts/models.md [hunk 1]
```diff
@@ -11,20 +11,33 @@ Models share many similarities with Python's dataclasses, but have been designed
 differences that streamline certain workflows related to validation, serialization, and JSON schema generation.
 You can find more discussion of this in the [Dataclasses](dataclasses.md) section of the docs.
 
-Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
+Untrusted data can be passed to a model and, after validation, Pydantic guarantees that the fields
```
Comment (samuelcolvin) [inline docs/concepts/models.md:14]: i think this was fine since we do do parsing - e.g. of JSON.

File: docs/concepts/models.md [hunk 2]
```diff
@@ -11,20 +11,33 @@ Models share many similarities with Python's dataclasses, but have been designed
...
 of the resultant model instance will conform to the field types defined on the model.
 
 !!! note
-    Pydantic is primarily a parsing and transformation library, **not a validation library**.
-    Validation is a means to an end: building a model which conforms to the types and constraints provided.
+    The use of the terms "parse" and "validate" within Pydantic's context has been a source of confusion.
```
Comment (samuelcolvin) [inline docs/concepts/models.md:18]: I don't think we need the big about "parse" and "validate", instead we should just explain why we use the word "validate".

File: docs/concepts/models.md [hunk 3]
```diff
@@ -11,20 +11,33 @@ Models share many similarities with Python's dataclasses, but have been designed
 differences that streamline certain workflows related to validation, serialization, and JSON schema generation.
 You can find more discussion of this in the [Dataclasses](dataclasses.md) section of the docs.
 
-Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
+Untrusted data can be passed to a model and, after validation, Pydantic guarantees that the fields
 of the resultant model instance will conform to the field types defined on the model.
 
 !!! note
```
Comment (samuelcolvin) [inline docs/concepts/models.md:17]:
```suggestion
!!! note "Validation - deliberate misnomer"
```
or something?

File: docs/concepts/models.md [hunk 4]
```diff
@@ -11,20 +11,33 @@ Models share many similarities with Python's dataclasses, but have been designed
...
-    Pydantic is primarily a parsing and transformation library, **not a validation library**.
-    Validation is a means to an end: building a model which conforms to the types and constraints provided.
+    The use of the terms "parse" and "validate" within Pydantic's context has been a source of confusion.
+    While these terms were previously used interchangeably, moving forward, we aim to exclusively employ "validate",
+    with "parse" reserved specifically for discussions related to [JSON parsing](../concepts/json.md).
 
-    In other words, Pydantic guarantees the types and constraints of the output model, not the input data.
+    The potential confusion arises from the fact that, strictly speaking, Pydantic's primary focus
```
Comment (samuelcolvin) [inline docs/concepts/models.md:22]: I think we want to say earlier on that "validation" is "the most widely understood colloquial term for the task pydantic performs" or similar.

Generally we should sound more like "this is the state of the world, we're not fighting it" rather than "we made a mistake long ago and we're sticking to that mistake.

Comment (samuelcolvin) [reply docs/concepts/models.md:22]: You could say something like "for brevity we use the term "validation" to refer to the process of optional coercion and validation which Pydantic is known for".?

File: docs/concepts/models.md [hunk 5]
```diff
@@ -14,17 +14,38 @@ You can find more discussion of this in the [Dataclasses](dataclasses.md) sectio
 Untrusted data can be passed to a model and, after parsing and validation, Pydantic guarantees that the fields
 of the resultant model instance will conform to the field types defined on the model.
 
-!!! note
-    Pydantic is primarily a parsing and transformation library, **not a validation library**.
-    Validation is a means to an end: building a model which conforms to the types and constraints provided.
+!!! note "Validation - a _deliberate_ misnomer"
```
Comment (dmontagu) [inline docs/concepts/models.md:17]:
```suggestion
!!! note "Validation — a _deliberate_ misnomer"
```

Comment (dmontagu) [reply docs/concepts/models.md:17]: Em dash

File: docs/concepts/models.md [hunk 6]
```diff
@@ -14,17 +14,38 @@ You can find more discussion of this in the [Dataclasses](dataclasses.md) sectio
...
+    ### TL;DR
+
+    We use the term "validation" to refer to the process of constructing a model that adheres to specified types and
+    constraints. This task, which Pydantic is well known for, is most widely recognized as "validation" in colloquial terms.
```
Comment (dmontagu) [inline docs/concepts/models.md:21]:
```suggestion
    constraints. This task, which Pydantic is well known for, is most widely recognized as "validation" in colloquial terms,
    even though in other contexts the term "validation" may be more restrictive.
```
can take it or leave it

File: docs/concepts/models.md [hunk 7]
```diff
@@ -14,17 +14,39 @@ You can find more discussion of this in the [Dataclasses](dataclasses.md) sectio
...
+    ### The long version
+
+    The potential confusion around the term "validation" arises from the fact that, strictly speaking, Pydantic's
+    primary focus doesn't align precisely with the dictionary definition of "validation":
+
+    > ### validation
+    > _noun_
+    > the action of checking or proving the validity or accuracy of something.
+
+    In Pydantic, the term "validation" refers to the process of constructing a model that adheres to specified
```
Comment (samuelcolvin) [inline docs/concepts/models.md:35]:
```suggestion
    In Pydantic, the term "validation" refers to the process of instantiate a model (or other type) that adheres to specified
```

**Review by samuelcolvin — APPROVED** [review]
otherwise LGTM. @dmontagu might be good for you to look too.

File: docs/concepts/models.md (hunk 7 above)
Comment (sydney-runkle) [reply docs/concepts/models.md]: Good call. Changed to "instantiating a model (or other type)" both here and in the TL;DR section.
